app/models/tweet.rb

```diff
@@ -0,0 +1,14 @@
+class Tweet < ApplicationRecord
+  belongs_to :user
+  #self reference
+  belongs_to :retweet, class_name: "Tweet", optional: true
+  #a Tweet can have many replies
+  has_many :replies
+  #a bookmark belongs to a single Tweet
+  has_one :bookmarks
```

Comment:
A tweet can be bookmarked by multiple users and at the same time a user can bookmark multiple tweets, `has_many` makes more sense for this releationship

Reply:
I didn’t take it in consideration. Thank you!